pkg/agent/agent.go

```diff
@@ -543,60 +495,40 @@ func (a *Agent) watch( //nolint:gocognit,cyclop // TODO: Refactor this function
 			a.ReportSnapshot(ctx)
 		} else {
 			// Don't report if the Director told us to stop reporting, if we are
 			// already sending a report or waiting for the minimum time between
 			// reports, or if there is nothing new to report right now.
-			dlog.Debugf(ctx, "Not reporting snapshot [reporting stopped = %t] [report running = %t] [report to send is nil = %t]",
-				a.reportingStopped, a.reportRunning.Value(), (a.reportToSend == nil))
+			dlog.Tracef(ctx, "Not reporting snapshot [reporting stopped = %t] [report running = %t] [report to send is nil = %t]",
+				a.reportingStopped, a.reportRunning.Load(), a.reportToSend == nil)
 		}
 
 		// only get diagnostics and metrics from edgissary if it is present
 		// TODO get metrics/diagnostics from traffic manager?
 		if !a.emissaryPresent {
-			dlog.Debugf(ctx, "Edgissary not present, not reporting edgissary diagnostics and metrics")
+			dlog.Tracef(ctx, "Edgissary not present, not reporting edgissary diagnostics and metrics")
 			continue
 		}
 
-		if !a.diagnosticsReportingStopped && !a.diagnosticsReportRunning.Value() && a.reportDiagnosticsAllowed {
-			a.ReportDiagnostics(ctx, diagnosticsURL, ambHost)
+		if !a.diagnosticsReportingStopped && !a.diagnosticsReportRunning.Load() && a.reportDiagnosticsAllowed {
+			a.ReportDiagnostics(ctx, a.AESDiagnosticsURL)
 		} else {
 			// Don't report if the Director told us to stop reporting, if we are
 			// already sending a report or waiting for the minimum time between
 			// reports
-			dlog.Debugf(ctx, "Not reporting diagnostics [reporting stopped = %t] [report running = %t]", a.diagnosticsReportingStopped, a.diagnosticsReportRunning.Value())
+			dlog.Tracef(ctx, "Not reporting diagnostics [reporting stopped = %t] [report running = %t]", a.diagnosticsReportingStopped, a.diagnosticsReportRunning.Load())
 		}
 
-		if !a.reportingStopped && !a.metricsReportRunning.Value() {
+		if !a.reportingStopped && !a.metricsReportRunning.Load() {
 			a.ReportMetrics(ctx)
 		} else {
-			dlog.Debugf(ctx, "Not reporting diagnostics [reporting stopped = %t] [report running = %t]", a.reportingStopped, a.metricsReportRunning.Value())
+			dlog.Tracef(ctx, "Not reporting diagnostics [reporting stopped = %t] [report running = %t]", a.reportingStopped, a.metricsReportRunning.Load())
 		}
 	}
 }
 
-func (a *Agent) parseCommAddr() error {
-	newConnInfo, err := connInfoFromAddress(a.connAddress)
-	if err != nil {
-		return err
-	}
-
-	if a.connInfo == nil || *newConnInfo != *a.connInfo {
-		// The configuration for the Director endpoint has changed: either this
-		// is the first snapshot or the user changed the value.
-		//
-		// Close any existing communications channel so that we can create
-		// a new one with the new endpoint.
-		a.ClearComm()
-
-		// Save the new endpoint information.
-		a.connInfo = newConnInfo
-	}
-	return nil
-}
-
 func (a *Agent) handleAmbassadorEndpointChange(ctx context.Context, ambassadorHost string) {
```

Comment:
FYI, currently we only monitor the namespace the agent is in for emissary endpoints, meaning if emissary is in a different namespace, it will not be detected. This is fine to leave out for now, but If we ever want mappings for emissary (like for ingressInfo), we should change this